prograils/lit
# test suite updates

Admittedly there's a bunch of yak shaving in this PR, but as I was attempting to accomplish a handful of things:

1. shore up the gem's true dependencies in gemspec
2. get travis ci working
3. use latest rails versions in appraisal
4. establish predictable steps to run the test suite
5. eliminate the noisy output in the test suite
6. generally understand the test suite & appraisal's setup
7. Add Gemfile and Gemfile.lock for gem reqs (appraisal, etc)

I mostly struck out on #5.  

#6 causes the error I mentioned here - https://github.com/prograils/lit/issues/117#issuecomment-442908214 which must be some weird appraisal/BUNDLE_GEMFILE conflict.  I wasn't able to straighten that out.  

Some other things along the way:
1. modernized how the suite (and the dummy app) load gems through boot.rb, environment.rb, application.rb (I updated these from a newly generated gem and rails app) (see #5 above)
2. start tracking gemfile.lock files since that's generally a best practice
3. dropped a bunch of things that weren't getting used
4. dropped user/pass req on test DB which should eliminate the need for a special local pg user

I realize there's a bunch of changes here.  Let me know if the new steps in the README work well for you.  They're working well over here.  Please let me know if you'd like me to unwind some of these things.

Thanks!
